### PR TITLE
feat(toolchain): add python 3.13.0, updated other versions and bumped coverage to 7.6.1

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Added python 3.13.0 toolchain
- Updated other versions in the toolchain
- Bumped coverage version to 7.6.1
- Updated CHANGELOG.md

### CODE REVIEW

File: python/versions.bzl
Comment (aignas) [inline python/versions.bzl:500]: Please update the rest of the toolchains to use the same indygreg release. If the same version is already present in the `TOOL_VERSIONS`, then we should overwrite the old values with the new values.

**Review by aignas — CHANGES_REQUESTED** [review]

File: python/versions.bzl
Comment (vfdev-5) [reply python/versions.bzl:500]: @aignas you mean to update for all other python versions like 3.12.X etc ?
I wonder whether there is an automatic way to get sha256 hashes for the versions? For 3.13 I did that manually, downloading tar.gz and running `sha256sum`...

**Comment by aignas** [comment]
If you run the target defined in https://github.com/bazelbuild/rules_python/blob/413690fd09b068ab43ec9c1d37d20d8017747f2c/python/private/BUILD.bazel#L509 then rules_python will download all of the toolchains and print the checksums. The only thing you need to have is the list of empty platform identifiers, like:
```starlark
    "3.13.0": {
        "url": "20240909/cpython-{python_version}rc2+20240909-{platform}-{build}.tar.gz",
        "sha256": {
            "aarch64-apple-darwin": "",
            "aarch64-unknown-linux-gnu": "",
            "ppc64le-unknown-linux-gnu": "",
            "s390x-unknown-linux-gnu": "",
            "x86_64-apple-darwin": "",
            "x86_64-pc-windows-msvc": "",
            "x86_64-unknown-linux-gnu": "",
        },
        "strip_prefix": "python",
    },
```

Sorry for being late here.

I think you also need to change any reference to `MINOR_MAPPING`, which is using the values in the [folowing way](https://github.com/bazelbuild/rules_python/blob/413690fd09b068ab43ec9c1d37d20d8017747f2c/WORKSPACE#L49). I think it is the only place. Adding a new Python version is always slightly annoying.

File: CHANGELOG.md
Comment (aignas) [inline CHANGELOG.md:47]: I think this part of the change is lost somewhere in the `git` dungeons. Could you please check if you have pushed it?

**Comment by vfdev-5** [comment]
@aignas thanks for the review! I agree there was something weird in the last commit in addition to a merge conflict with CHANGELOG.md. I fixed it and now it should be good.

File: python/versions.bzl
Comment (aignas) [inline python/versions.bzl:564]: Since the official release of 3.13.0 has been cut, I think we can wait for indigreg release to crop up once again before merging this. That will save some back-and-forth.

**Comment by aignas** [comment]
It seems that they were really fast: https://github.com/indygreg/python-build-standalone/releases/tag/20241008 contains the new builds and even the free-threading ones. I think adding the freethreading values in the `TOOL_VERSIONS` should be done in #2129.

**Comment by aignas** [comment]
Could you also please add new binaries for `coverage.py`? This is needed since it is a new major version of python. I think just upgrading to the latest version (https://pypi.org/project/coverage/#files) should be sufficient. The docs on how to do it are in `DEVELOPING.md` file.

Feel free to leave notes there about how to add a new minor python version.

**Comment by vfdev-5** [comment]
> Could you also please add new binaries for `coverage.py`? This is needed since it is a new major version of python. I think just upgrading to the latest version (https://pypi.org/project/coverage/#files) should be sufficient. The docs on how to do it are in `DEVELOPING.md` file.
> 
> Feel free to leave notes there about how to add a new minor python version.

I'm not very sure how this works, I updated the PR, please review

**Review by aignas — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
